File: nest/v1/backend-ms1-new/src/app/module/service/admin/commun/ProduitAdminServiceImpl.ts
import {Injectable, NotFoundException} from "@nestjs/common";

import {AbstractServiceImpl} from "src/app/zynerator/service/AbstractServiceImpl";

import {ProduitDao} from "src/app/controller/dao/facade/core/commun/ProduitDao";
import {ProduitCriteria} from "src/app/controller/dao/criteria/core/commun/ProduitCriteria";
import {Produit} from "src/app/controller/bean/core/commun/Produit";
import {ProduitDto} from "src/app/controller/dto/commun/ProduitDto";


@Injectable()
export class ProduitAdminServiceImpl extends AbstractServiceImpl<Produit, ProduitCriteria, ProduitDao>{

    constructor(private readonly dao: ProduitDao ,
    ) {
        super(dao);
    }

    async save(item: Produit): Promise<Produit> {
        const saved = await this.dao.save(item);
        return saved;
    }


    async update(item: Produit): Promise<Produit> {
        const saved = await this.dao.saveOrUpdate(item);
        return saved;
    }

    async updateMultiple(items: Produit[]): Promise<void> {
        if (items) {
            items.forEach(e => this.update(e))
        }
    }

    async  findAllOptimized(): Promise<ProduitDto[]> {
        return this.dao.findAllOptimized();
    }

    async findAll(): Promise<Produit[]> {
        return this.dao.findAll();
    }

    async findById(id: number): Promise<Produit> {
        return this.dao.findById(id);
    }


    async deleteById(id: number): Promise<number> {
        const existing = await this.findById(id);
        if (!existing) {
            throw new NotFoundException(`Produit with ID ${id} not found.`);
        }
        const result = await this.dao.deleteById(existing.id);
        return result;
    }

    async deleteMultiple(items: Produit[]): Promise<Produit[]> {
        const deletedItems: Produit[] = [];
        for (const item of items) {
            await this.deleteById(item.id);
            deletedItems.push(item);
        }
        return deletedItems;
    }


    async findWithAssociatedLists(id: number): Promise<Produit> {
        const result = await this.dao.findById(id);
    return result;
    }

    async updateWithAssociatedLists(item: Produit): Promise<Produit> {
         return await this.dao.saveOrUpdate(item);
    }
}

File: spring-angular/roma/v4/backend-ms1/src/main/java/ma/zs/stocky/service/impl/admin/commun/ProduitAdminServiceImpl.java
package ma.zs.stocky.service.impl.admin.commun;


import ma.zs.stocky.bean.core.commun.Produit;
import ma.zs.stocky.dao.criteria.core.commun.ProduitCriteria;
import ma.zs.stocky.dao.facade.core.commun.ProduitDao;
import ma.zs.stocky.dao.specification.core.commun.ProduitSpecification;
import ma.zs.stocky.service.facade.admin.commun.ProduitAdminService;
import ma.zs.stocky.zynerator.service.AbstractServiceImpl;
import ma.zs.stocky.zynerator.util.ListUtil;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.ArrayList;





import org.springframework.beans.factory.annotation.Autowired;


import java.util.List;
@Service
public class ProduitAdminServiceImpl extends AbstractServiceImpl<Produit, ProduitCriteria, ProduitDao> implements ProduitAdminService {






    public Produit findByReferenceEntity(Produit t){
        return  dao.findByReference(t.getReference());
    }


    public List<Produit> findAllOptimized() {
        return dao.findAllOptimized();
    }





    public void configure() {
        super.configure(Produit.class, ProduitSpecification.class);
    }



    public ProduitAdminServiceImpl(ProduitDao dao) {
        super(dao);
    }

}
